### PR TITLE
feat: enable parallel feature for `cc`

### PR DESCRIPTION
**Pull request by rumblefrog**
Enables parallel feature for `cc`

### CODE REVIEW

File: cpp_build/Cargo.toml [hunk 1]
```diff
@@ -18,8 +18,8 @@ docs-only = []
 
 [dependencies]
 lazy_static = "1.0"
-cc = "1.0.38"
-cpp_common = { path = "../cpp_common", version = "=0.5.7" }
+cc = { version = "1.0", features = ["parallel"] }
```
Comment (ogoffart) [inline cpp_build/Cargo.toml:21]: why did you remove the .38?   this crate won't work with older version of `cc`

**Review by ogoffart — APPROVED** [review]
Thanks for your patch!

File: cpp_build/Cargo.toml (hunk 1 above)
Comment (rumblefrog) [reply cpp_build/Cargo.toml:21]: I was unaware it does not work with older versions of cc. I'm currently using my branch and have not seen any regression. Could you explain the .38 pin?

**Comment by ogoffart** [comment]
that said, what's the motivation to enable this feature?
The cpp-build crate only build one file.  I don't think this would have any impact, would it?
Why is it not enabled in by default in cc?

File: cpp_build/Cargo.toml (hunk 1 above)
Comment (ogoffart) [reply cpp_build/Cargo.toml:21]: this is not a pin.  With cargo, this is equivalent to "^1.0.38" which means later version are also accepted.
It needs the .38 because it uses some API that is only available in that version: (eg as explained in https://github.com/mystor/rust-cpp/commit/40f33243219fb7d91d65bf66c124b3eef3bf1042 )
So this is avoiding the (admittedly unlikely at this point) case if someone had a very old Cargo.lock with a very old version of cc, and adds the cpp-build crate.

**Comment by rumblefrog** [comment]
> that said, what's the motivation to enable this feature? The cpp-build crate only build one file. I don't think this would have any impact, would it? Why is it not enabled in by default in cc?

cpp-build exposes `cpp_build::Config`, and I'm currently adding a list of additional cpp files to be compile with it via `Config::file`.

CC Parallel vs Non-Parallel `compile_objects`:

https://github.com/rust-lang/cc-rs/blob/cf0a78b8ba8d15301c6596b35842da38f62b2233/src/lib.rs#L1201

https://github.com/rust-lang/cc-rs/blob/cf0a78b8ba8d15301c6596b35842da38f62b2233/src/lib.rs#L1352

I'm uncertain why this is not enabled by default, but I could revise this PR to gate it behind a feature flag as well.

**Comment by rumblefrog** [comment]
I have gated the parallel feature of cc.

**Comment by ogoffart** [comment]
> I'm currently adding a list of additional cpp files to be compile with it via Config::file.

Indeed, makes sense.

> I have gated the parallel feature of cc.

Thanks, then we don't depend on that dependency if not needed.
